zeebe/snapshot/src/test/java/io/camunda/zeebe/snapshots/transfer/SnapshotTransferTest.java

```diff
@@ -53,12 +58,13 @@ public void beforeEach() throws Exception {
 
     actorScheduler.submitActor(receiverSnapshotStore);
     actorScheduler.workUntilDone();
-    final ConcurrencyControl concurrencyControl = senderSnapshotStore;
-    final SnapshotTransferService transferService =
-        new SnapshotTransferServiceImpl(
-            senderSnapshotStore, 1, SnapshotCopyUtil.copyAllFiles(), concurrencyControl);
     snapshotTransfer =
-        new SnapshotTransfer(transferService, receiverSnapshotStore, concurrencyControl);
+        new SnapshotTransferImpl(
+            control ->
+                new SnapshotTransferServiceImpl(
+                    senderSnapshotStore, 1, SnapshotCopyUtil.copyAllFiles(), control),
+            receiverSnapshotStore);
+    actorScheduler.submitActor(snapshotTransfer);
```

Comment:
:x: This should be scheduled as an io-bound actor

Reply:
:x: Also seems like we just leak this actor. Or did I miss a close call somewhere?

Reply:
Oh it's just tests, nevermind then.

Reply:
:thinking: all of them in the tests probably.
Actually I think all actors in the snapshot test module are not scheduled on an IO scheduler. 

:thought_balloon: the scheduling hint should be part of the `Actor` interface, so there's no risk on scheduling an actor on the wrong scheduler

Reply:
In the tests they are run in `ControlledActorSchedulerExtension`, so I don't think that really matters :shrug: